packages/contracts-bedrock/test/L2/OptimismSuperchainERC20Factory.t.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+// Testing utilities
+import { Test } from "forge-std/Test.sol";
+import { EIP1967Helper } from "test/mocks/EIP1967Helper.sol";
+
+// Libraries
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { CREATE3, Bytes32AddressLib } from "@rari-capital/solmate/src/utils/CREATE3.sol";
+import { IBeacon } from "@openzeppelin/contracts-v5/proxy/beacon/IBeacon.sol";
+
+// Target contract
+import { OptimismSuperchainERC20Factory, OptimismSuperchainERC20 } from "src/L2/OptimismSuperchainERC20Factory.sol";
+
+/// @title OptimismSuperchainERC20FactoryTest
+/// @notice Contract for testing the OptimismSuperchainERC20Factory contract.
+contract OptimismSuperchainERC20FactoryTest is Test {
+    using Bytes32AddressLib for bytes32;
+
+    OptimismSuperchainERC20 public superchainERC20Impl;
+    OptimismSuperchainERC20Factory public superchainERC20Factory;
+
+    /// @notice Sets up the test suite.
+    function setUp() public {
+        superchainERC20Impl = new OptimismSuperchainERC20();
+
+        // Deploy the OptimismSuperchainERC20Beacon contract
+        _deployBeacon();
+
+        superchainERC20Factory = new OptimismSuperchainERC20Factory();
+    }
+
+    /// @notice Deploy the OptimismSuperchainERC20Beacon predeploy contract
+    function _deployBeacon() internal {
+        // Deploy the OptimismSuperchainERC20Beacon implementation
+        address _addr = Predeploys.OPTIMISM_SUPERCHAIN_ERC20_BEACON;
+        address _impl = Predeploys.predeployToCodeNamespace(_addr);
+        vm.etch(_impl, vm.getDeployedCode("OptimismSuperchainERC20Beacon.sol:OptimismSuperchainERC20Beacon"));
+
+        // Deploy the ERC1967Proxy contract at the Predeploy
+        bytes memory code = vm.getDeployedCode("universal/Proxy.sol:Proxy");
+        vm.etch(_addr, code);
+        EIP1967Helper.setAdmin(_addr, Predeploys.PROXY_ADMIN);
+        EIP1967Helper.setImplementation(_addr, _impl);
+
+        // Mock implementation address
+        vm.mockCall(
+            _impl, abi.encodeWithSelector(IBeacon.implementation.selector), abi.encode(address(superchainERC20Impl))
+        );
+    }
+
+    /// @notice Test that calling `deploy` with valid parameters succeeds.
+    function test_deploy_succeeds(
+        address _caller,
+        address _remoteToken,
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals
+    )
+        public
+    {
+        // Arrange
+        bytes32 salt = keccak256(abi.encode(_remoteToken, _name, _symbol, _decimals));
+        address deployment = _calculateTokenAddress(salt, address(superchainERC20Factory));
+
+        vm.expectEmit(address(superchainERC20Factory));
```

Comment:
should this have a true, true statement for the indexed arguments?

Reply:
i started changing them all after this comment https://github.com/ethereum-optimism/optimism/pull/11479#discussion_r1725659041

Reply:
Excellent, was not aware of this